use-ssh/cleanup.ts

```diff
@@ -3,7 +3,11 @@ import { exec } from "@actions/exec";
 
 async function run() {
   try {
-    await exec("ssh-agent -k");
+    process.env.SSH_AUTH_SOCK = core.getState("SSH_AUTH_SOCK");
+    process.env.SSH_AGENT_PID = core.getState("SSH_AGENT_PID");
+    const sshKey = core.getState("SSH_KEY");
+    core.setSecret(sshKey);
+    await exec(`echo ${sshKey} | ssh-add -d -`);
```

Comment:
With the new version, only the `ssh-key` that passed as input is deleted from the agent